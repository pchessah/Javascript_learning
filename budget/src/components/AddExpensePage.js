import React from 'react'

const AddExpensePage = () => (
    <div>
        This is from my add expense componenent
    </div>
);

export default AddExpensePage;