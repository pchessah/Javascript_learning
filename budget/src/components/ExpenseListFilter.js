import React from 'react'
import { connect } from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount } from '../action/filters'

const ExpenseListFilters = (props) => {
     return (
        <div>
            <input value = {props.filters.text} onChange ={(e) => {
                props.dispatch(setTextFilter(e.target.value))
            }} />
            <select
            value = {props.filters.sortBy}
            onChange = {(e) => {
                if (e.target.value === 'date'){
                    props.dispatch(sortByDate());
                } else if (e.target.value === 'amount') {
                    props.dispatch(sortByAmount());
                }
            }}>
                <option value = "date"> Date </option>
                <option value = "amount"> amount </option>
            </select>
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps)(ExpenseListFilters)