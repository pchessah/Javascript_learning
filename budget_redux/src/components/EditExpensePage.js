import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {startRemoveExpense, startEditExpense} from '../actions/expenses';

const EditExpensePage = (props) => {
    return (
        <div>
            <ExpenseForm
                expense={props.expense}
                onSubmit={(expense) => {
                    props.dispatch(startEditExpense(props.expense.id, expense));
                    props.history.push('/');
                }}            
            />
            <button onClick={() => {
                props.dispatch(startRemoveExpense({ id: props.expense.id}));
                props.history.push('/');
            }}>Remove</button>
        </div>
    );
};
const mapStateToProps = (state, ownProps) => {
    return {
        expense: state.expenses.find((expense) => expense.id === ownProps.match.params.id)
    }
}

export default connect(mapStateToProps)(EditExpensePage);