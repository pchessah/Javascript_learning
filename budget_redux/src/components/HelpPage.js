import React from 'react';

const HelpPage = () => (
    <div>
        This is from my Help Page componenent
    </div>
);

export default HelpPage;