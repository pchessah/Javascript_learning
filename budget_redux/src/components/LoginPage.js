import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'

export const LoginPage = ( {startLogin}) => {
    return (
        <div>
            <h1> Press Button to login with Google Account </h1>
            <button onClick = { startLogin}> Log In </button>
        </div>
    )
}

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect (null, mapDispatchToProps)(LoginPage)