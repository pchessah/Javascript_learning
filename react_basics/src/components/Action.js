import React from 'react-dom'

const Action = (props) => (
    <button 
            onClick = {props.handlePick}
            disabled = {!props.hasOption}
            > What should I do? </button>
)

export default Action;