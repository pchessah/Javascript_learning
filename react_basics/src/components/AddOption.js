import React from 'react'

//creating a class component
class AddOption extends React.Component {
    constructor(props) { //initialization of the object
        super (props); //super method must be passed with initialization
        this.handleAddOption = this.handleAddOption.bind(this); //binding the events
        this.state = { //setting the default state
            error: null 
        }
    }
    handleAddOption (e) {
        e.preventDefault ();

        const option = e.target.elements.option.value.trim();
        const data = this.props.handleData (option);

       this.setState (() => {
           return {
               error: data
           }
       })
        e.target.elements.option.value = '';
    }
    render() {
        return(
            <div>
                {this.state.error && <p className = "add-option-error"> {this.state.error}</p>}
               <form className= "add-option" onSubmit={this.handleAddOption}>
                   <input className = "add-option__input" type = "text" name = "option"/>
                   <button> Add Option </button>
               </form> 
            </div>
        );
    }
}

export default AddOption;