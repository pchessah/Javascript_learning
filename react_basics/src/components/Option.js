import React from 'react';

const Option = (props) => (
    <div>
    <div className = "option"> 
    <p className= "option_text">{props.count}.{props.optionText} </p>
    </div>
    <button className="button button--link"
    onClick = {(e) => {
        props.handleRemove(props.optionText)
    }}>  remove  </button>
    </div>
)

export default Option;
