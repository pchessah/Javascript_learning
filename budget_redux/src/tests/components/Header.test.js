import React from 'react'
import { shallow } from  'enzyme'
import {Header} from '../../components/Header'

test ('Should render Header page', () =>{
    const wrapper = shallow (<Header startLogout ={() => {}}/>)
    expect(wrapper).toMatchSnapshot();
})