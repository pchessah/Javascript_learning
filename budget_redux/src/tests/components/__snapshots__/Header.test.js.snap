// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header page 1`] = `
<header>
  <h1>
    Budget
  </h1>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    exact={true}
    to="/"
  >
           |Dashboard|    
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/create"
  >
          Create Expense|     
  </NavLink>
  <NavLink
    activeClassName="is-active"
    aria-current="page"
    to="/help"
  >
        Help|     
  </NavLink>
  <button
    onClick={[Function]}
  >
     Log Out 
  </button>
</header>
`;
